src/com/company/MyLinkedList.java

```diff
@@ -86,4 +86,27 @@ public void deleteNode(LinkedListNode node){
         LinkedListNode toBeDeleted = temp.getNextNode();
         temp.setNextNode(toBeDeleted.getNextNode());
     }
+    
+    public void insertAfter(LinkedListNode nodeAfterWhichIsInserted, LinkedListNode toBeInserted){
+        if(!exists(nodeAfterWhichIsInserted)){
+            return;
+        }
+
+        LinkedListNode temp = root;
+        LinkedListNode nodeBeforeWhichIsInserted = null;
+
+        while (!temp.equals(nodeAfterWhichIsInserted)) {
+            temp = temp.getNextNode();
+        }
+
+        if(temp.getNextNode()!=null){
+               nodeBeforeWhichIsInserted = temp.getNextNode();
+            }
+
+            temp.setNextNode(toBeInserted);
+
+        if(nodeBeforeWhichIsInserted != null){
```

Comment:
this if condition is redundant because of check on line 102. Also even if nextNode were null, nothing wrong would have happened from overwriting null with null.